File: src/main/java/no/nav/fo/veilarbregistrering/db/RepositoryConfig.java
package no.nav.fo.veilarbregistrering.db;

import no.nav.fo.veilarbregistrering.arbeidssoker.ArbeidssokerRepository;
import no.nav.fo.veilarbregistrering.db.arbeidssoker.ArbeidssokerRepositoryImpl;
import no.nav.fo.veilarbregistrering.db.oppgave.OppgaveRepositoryImpl;
import no.nav.fo.veilarbregistrering.db.profilering.ProfileringRepositoryImpl;
import no.nav.fo.veilarbregistrering.db.registrering.*;
import no.nav.fo.veilarbregistrering.oppgave.OppgaveRepository;
import no.nav.fo.veilarbregistrering.profilering.ProfileringRepository;
import no.nav.fo.veilarbregistrering.registrering.bruker.BrukerRegistreringRepository;
import no.nav.fo.veilarbregistrering.registrering.bruker.ReaktiveringRepository;
import no.nav.fo.veilarbregistrering.registrering.bruker.SykmeldtRegistreringRepository;
import no.nav.fo.veilarbregistrering.registrering.manuell.ManuellRegistreringRepository;
import no.nav.fo.veilarbregistrering.registrering.tilstand.RegistreringTilstandRepository;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;

@Configuration
public class RepositoryConfig {

    @Bean
    ReaktiveringRepository reaktiveringRepository(NamedParameterJdbcTemplate db) {
        return new ReaktiveringRepositoryImpl(db);
    }

    @Bean
    SykmeldtRegistreringRepository sykmeldtRegistreringRepository(NamedParameterJdbcTemplate db) {
        return new SykmeldtRegistreringRepositoryImpl(db);
    }

    @Bean
    BrukerRegistreringRepository brukerRegistreringRepository(NamedParameterJdbcTemplate db) {
        return new BrukerRegistreringRepositoryImpl(db);
    }

    @Bean
    RegistreringTilstandRepository registreringTilstandRepository(NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
        return new RegistreringTilstandRepositoryImpl(namedParameterJdbcTemplate);
    }

    @Bean
    ArbeidssokerRepository arbeidssokerRepository(NamedParameterJdbcTemplate db) {
        return new ArbeidssokerRepositoryImpl(db);
    }

    @Bean
    OppgaveRepository oppgaveRepository(NamedParameterJdbcTemplate db) {
        return new OppgaveRepositoryImpl(db);
    }

    @Bean
    ProfileringRepository profileringRepository(NamedParameterJdbcTemplate db) {
        return new ProfileringRepositoryImpl(db);
    }

    @Bean
    ManuellRegistreringRepository manuellRegistreringRepository(NamedParameterJdbcTemplate db) {
        return new ManuellRegistreringRepositoryImpl(db);
    }
}

File: src/main/kotlin/no/nav/fo/veilarbregistrering/db/RepositoryConfig.kt
package no.nav.fo.veilarbregistrering.db

import no.nav.fo.veilarbregistrering.aktorIdCache.AktorIdCacheRepository
import no.nav.fo.veilarbregistrering.arbeidssoker.formidlingsgruppe.FormidlingsgruppeRepository
import no.nav.fo.veilarbregistrering.arbeidssoker.meldekort.MeldekortRepository
import no.nav.fo.veilarbregistrering.db.aktorIdCache.AktorIdCacheRepositoryImpl
import no.nav.fo.veilarbregistrering.db.arbeidssoker.FormidlingsgruppeRepositoryImpl
import no.nav.fo.veilarbregistrering.db.arbeidssoker.MeldekortRepositoryImpl
import no.nav.fo.veilarbregistrering.db.oppgave.OppgaveRepositoryImpl
import no.nav.fo.veilarbregistrering.db.profilering.ProfileringRepositoryImpl
import no.nav.fo.veilarbregistrering.db.registrering.*
import no.nav.fo.veilarbregistrering.oppgave.OppgaveRepository
import no.nav.fo.veilarbregistrering.profilering.ProfileringRepository
import no.nav.fo.veilarbregistrering.registrering.formidling.RegistreringTilstandRepository
import no.nav.fo.veilarbregistrering.registrering.ordinaer.BrukerRegistreringRepository
import no.nav.fo.veilarbregistrering.registrering.reaktivering.ReaktiveringRepository
import no.nav.fo.veilarbregistrering.registrering.sykmeldt.SykmeldtRegistreringRepository
import no.nav.fo.veilarbregistrering.registrering.veileder.ManuellRegistreringRepository
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate

@Configuration
class RepositoryConfig {
    @Bean
    fun reaktiveringRepository(db: NamedParameterJdbcTemplate): ReaktiveringRepository {
        return ReaktiveringRepositoryImpl(db)
    }

    @Bean
    fun sykmeldtRegistreringRepository(db: NamedParameterJdbcTemplate): SykmeldtRegistreringRepository {
        return SykmeldtRegistreringRepositoryImpl(db)
    }

    @Bean
    fun brukerRegistreringRepository(db: NamedParameterJdbcTemplate): BrukerRegistreringRepository {
        return BrukerRegistreringRepositoryImpl(db)
    }

    @Bean
    fun registreringTilstandRepository(namedParameterJdbcTemplate: NamedParameterJdbcTemplate): RegistreringTilstandRepository {
        return RegistreringTilstandRepositoryImpl(namedParameterJdbcTemplate)
    }

    @Bean
    fun arbeidssokerRepository(db: NamedParameterJdbcTemplate): FormidlingsgruppeRepository {
        return FormidlingsgruppeRepositoryImpl(db)
    }

    @Bean
    fun meldekortRepository(db: NamedParameterJdbcTemplate): MeldekortRepository {
        return MeldekortRepositoryImpl(db)
    }

    @Bean
    fun oppgaveRepository(db: NamedParameterJdbcTemplate): OppgaveRepository {
        return OppgaveRepositoryImpl(db)
    }

    @Bean
    fun profileringRepository(db: NamedParameterJdbcTemplate): ProfileringRepository {
        return ProfileringRepositoryImpl(db)
    }

    @Bean
    fun manuellRegistreringRepository(db: NamedParameterJdbcTemplate): ManuellRegistreringRepository {
        return ManuellRegistreringRepositoryImpl(db)
    }

    @Bean
    fun aktorIdCacheRepository(db: NamedParameterJdbcTemplate): AktorIdCacheRepository {
        return AktorIdCacheRepositoryImpl(db)
    }
}
